static/analysis.go

```diff
@@ -83,53 +87,97 @@ func analyzeIdentityCall(c *ssa.Call) configkit.Identity {
+// arguments. It assumes that the handler is always the first argument.
+//
+// If the first argument is not a pointer to ssa.MakeInterface instruction, this
+// function has no affect.
+func addHandlerFromArguments(
 	prog *ssa.Program,
-	typ types.Type,
+	args []ssa.Value,
+	hs configkit.HandlerSet,
 	ht configkit.HandlerType,
-) configkit.Handler {
-	hdr := &entity.Handler{
-		HandlerTypeValue: ht,
-		TypeNameValue:    typ.String(),
-		MessageNamesValue: configkit.EntityMessageNames{
-			Produced: message.NameRoles{},
-			Consumed: message.NameRoles{},
-		},
-	}
+) {
+	if mi, ok := args[0].(*ssa.MakeInterface); ok {
+		typ := mi.X.Type()
+
+		hdr := &entity.Handler{
+			HandlerTypeValue: ht,
+			TypeNameValue:    typ.String(),
+			MessageNamesValue: configkit.EntityMessageNames{
+				Produced: message.NameRoles{},
+				Consumed: message.NameRoles{},
+			},
+		}
 
-	pkg := pkgOfNamedType(typ)
-	fn := prog.LookupMethod(typ, pkg, "Configure")
+		pkg := pkgOfNamedType(typ)
+		fn := prog.LookupMethod(typ, pkg, "Configure")
 
-	for _, c := range findConfigurerCalls(fn) {
-		args := c.Common().Args
+		for _, c := range findConfigurerCalls(fn) {
+			args := c.Common().Args
 
-		switch c.Common().Method.Name() {
-		case "Identity":
-			hdr.IdentityValue = analyzeIdentityCall(c)
-		case "ConsumesCommandType":
-			if mi, ok := args[0].(*ssa.MakeInterface); ok {
-				hdr.MessageNamesValue.Consumed.Add(
-					message.NameFromType(
-						mi.X.Type(),
-					),
+			switch c.Common().Method.Name() {
+			case "Identity":
+				hdr.IdentityValue = analyzeIdentityCall(c)
+			case "ConsumesCommandType":
+				addMessageFromArguments(
+					args,
+					hdr.MessageNamesValue.Consumed,
 					message.CommandRole,
 				)
-			}
-		case "ProducesEventType":
-			if mi, ok := args[0].(*ssa.MakeInterface); ok {
-				hdr.MessageNamesValue.Produced.Add(
-					message.NameFromType(
-						mi.X.Type(),
-					),
+			case "ConsumesEventType":
+				addMessageFromArguments(
+					args,
+					hdr.MessageNamesValue.Consumed,
+					message.EventRole,
+				)
+			case "ProducesCommandType":
+				addMessageFromArguments(
+					args,
+					hdr.MessageNamesValue.Produced,
+					message.CommandRole,
+				)
+			case "ProducesEventType":
+				addMessageFromArguments(
+					args,
+					hdr.MessageNamesValue.Produced,
 					message.EventRole,
 				)
+			case "SchedulesTimeoutType":
+				addMessageFromArguments(
+					args,
+					hdr.MessageNamesValue.Consumed,
+					message.TimeoutRole,
+				)
+				addMessageFromArguments(
+					args,
+					hdr.MessageNamesValue.Produced,
+					message.TimeoutRole,
+				)
 			}
 		}
+
+		hs.Add(hdr)
 	}
+}
 
-	return hdr
+// addMessageFromArguments adds a message with the given role r to the
+// name role map from the given arguments. It assumes that the message
+// is always the first argument.
+//
+// If the first argument is not a pointer to ssa.MakeInterface instruction, this
+// function has no affect.
```

Comment:
```suggestion
// addMessageFromArguments analyzes args to deduce the type of a message.
// It assumes that the message is always the first argument.
//
// If the first argument is not a pointer to ssa.MakeInterface instruction, this
// function has no effect; otherwise the message type is added to nr using the
// role given by r.
```